rcl_yaml_param_parser/src/parse.c

```diff
@@ -87,9 +88,12 @@ void * get_value(
   RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
     &allocator, "allocator is invalid", return NULL);
 
+  bool str_tag_enabled = tag != NULL && strcmp(YAML_STR_TAG, (char *)tag) == 0;
+
   /// Check if it is bool
   if (style != YAML_SINGLE_QUOTED_SCALAR_STYLE &&
-    style != YAML_DOUBLE_QUOTED_SCALAR_STYLE)
+    style != YAML_DOUBLE_QUOTED_SCALAR_STYLE &&
+    !str_tag_enabled)
   {
```

Comment:
Adding this check everywhere seems to be brittle.  If someone adds a new type, but forgets to add the `!str_tag_enabled`, it will fail in very specific circumstances.

Instead, can we just do the following at the top of the function:

```
if (tag != NULL && strcmp(YAML_STR_TAG, (char *)tag) == 0) {
  *val_type = DATA_TYPE_STRING;
  return rcutils_strdup(value, allocator);
}
```

Reply:
It seems a duplicated code [there](https://github.com/ros2/rcl/blob/6b516f0455c4739b64ff0dfa11387712ce6648d6/rcl_yaml_param_parser/src/parse.c#L212-L213), can we use `goto string_value;` instead?